src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { memo } from 'react';
+import cn from 'classnames';
+import { FilterStatus } from '../../types/FilterStatus';
+
+type Props = {
+  filterStatus: string;
+  onFilterStatusChange: React.Dispatch<React.SetStateAction<FilterStatus>>;
+};
+
+export const Filter: React.FC<Props> = memo((props) => {
+  const { filterStatus, onFilterStatusChange } = props;
+
+  return (
+    <nav className="filter" data-cy="Filter">
+      <a
+        data-cy="FilterLinkAll"
+        href="#/"
+        className={cn(
+          'filter__link',
+          { selected: filterStatus === FilterStatus.All },
+        )}
+        onClick={() => onFilterStatusChange(FilterStatus.All)}
```

Comment:
What will happen after `onClick`? -> `setFilterStatus`. But not `onFilterStatusChange`
```suggestion
        onClick={() => setFilterStatus(FilterStatus.All)}
```